server/base/src/main/java/org/apache/accumulo/server/master/balancer/HostRegexTableLoadBalancer.java

```diff
@@ -367,8 +396,30 @@ public long balance(SortedMap<TServerInstance,TabletServerStatus> current, Set<K
     }
 
     if (migrations != null && migrations.size() > 0) {
-      LOG.warn("Not balancing tables due to {} outstanding migrations", migrations.size());
-      return minBalanceTime;
+      if (migrations.size() >= maxOutstandingMigrations) {
+        LOG.warn("Not balancing tables due to {} outstanding migrations", migrations.size());
+        LOG.trace("Sample up to 10 outstanding migrations: {}", Iterables.limit(migrations, 10));
```

Comment:
Nope :) Think about calling a method in java. You still have to cast/convert the object into the correct parameters (e.g. the Iterable<String> into a String) regardless of whether or not it gets logged.

The thing that SLF4j loggers get you is that you don't build the log message if it's not going to be logged.